vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/dialog_parser.rb

```diff
@@ -0,0 +1,116 @@
+
+
+def vmdb_object_from_array_entry(entry)
+  model, id = entry.split("::")
+  $evm.vmdb(model, id.to_i) if model && id
+end
+
+def parent_task(task)
+  return task if task.miq_request_task.nil?
+  parent_task(task.miq_request_task)
+end
+
+def add_hash_value(sequence_id, option_key, value)
+  $evm.log("info", "Adding seq_id: #{sequence_id} key: #{option_key} value: #{value} ")
+  value
+end
+
+def process_comma_separated_object_array(sequence_id, option_key, value, hash)
+  return if value.nil?
+  options_value_array = []
+  value.split(",").each do |entry|
+    vmdb_obj = vmdb_object_from_array_entry(entry.to_s)
+    next if vmdb_obj.nil?
+    options_value_array << vmdb_obj.name
+  end
+  hash[sequence_id][option_key] = options_value_array
+end
+
+def option_hash_value(dialog_key, dialog_value, options_hash)
+  options_regex = /^dialog_option_(\d*)_(.*)/i
```

Comment:
@tinaafitz 
Can we use regexp to set the variables e.g
/^dialog_option_(?<sequence>\d*)_(?<option_key>.*)/i =~ dialog_key
options_hash[sequence.to_i][option_key.to_sym] = add_hash_value(sequence.to_i, option_key.to_sym, dialog_value)